obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+    # parse the header
+    #
+    # The sac header has 70 floats, 40 integers, then 192 bytes
+    #    in strings. Store them in array (and convert the char to a
+    #    list). That's a total of 632 bytes.
+    # --------------------------------------------------------------
+    # read in the float values
+    # TODO: use native '=' dtype byteorder instead of forcing little endian?
+    hf = np.array([i.split() for i in contents[:14]],
+                   dtype=native_str('<f4')).ravel()
+    # read in the int values
+    hi = np.array([i.split() for i in contents[14: 14 + 8]],
+                   dtype=native_str('<i4')).ravel()
+    # reading in the string part is a bit more complicated
+    # because every string field has to be 8 characters long
+    # apart from the second field which is 16 characters long
+    # resulting in a total length of 192 characters
+    hs = init_header_arrays('str')
+    for i, j in enumerate(range(0, 24, 3)):
+        line = contents[14 + 8 + i]
+        hs[j:j + 3] = np.fromstring(line, dtype=native_str('|S8'), count=3)
+    # --------------------------------------------------------------
+    # read in the seismogram points
+    # --------------------------------------------------------------
+    if headonly:
+        data = None
+    else:
+        data = np.array([i.split() for i in contents[30:]],
+                         dtype=native_str('<f4')).ravel()
+
+        npts = hi[HD.INTHDRS.index('npts')]
+        if len(data) != npts:
+            if is_file_name:
+                fh.close()
+            raise SacIOError("Cannot read all data points")
+
+    if is_file_name:
+        fh.close()
+
+    return hf, hi, hs, data
+
+
+def write_sac(dest, hf, hi, hs, data=None, byteorder=None):
+    """
+    Write the header and (optionally) data arrays to a SAC binary file.
+
+    Parameters
+    ----------
+    dest : str or File-like object
+        Full path or File-like object from SAC binary file on disk.
+        If data is None, file mode should be 'wb+'.
+    hf, hi, hs : numpy.ndarray
+        The float, integer, and string header arrays.
+    data : numpy.ndarray, optional
+        float32 data array.  If omitted or None, it is assumed that the user
+        intends to overwrite/modify only the header arrays of an existing file.
+        Equivalent to "writehdr".
+    byteorder : str {'little', 'big'}, optional
+        Desired output byte order.  If omitted, arrays are written as they are.
+        If data=None, better make sure the file you're writing to has the same
+        byte order as headers you're writing.
+
+    Notes
+    -----
+    A user can/should not _create_ a header-only binary file.  Use mode 'wb+'
+    for data=None (headonly) writing.
+
+    """
+
+    # deal with file name versus File-like object, and file mode
+    if data is None:
+        # file exists, just modify it (don't start from scratch)
+        fmode = 'rb+'
```

Comment:
I realize that this functionality adds complexity, and isn't quite ready yet, but modifying lots SAC headers without touching the data is very common in my experience.  Providing a way to muck with the headers en masse suggests that ObsPy(.io.sac...) may be used to prep data for the many research codes that use ancillary information in SAC headers, instead of using an installation of SAC itself to do that.  Again, I know that file I/O is ObsPy's main interest in an `io.sac` module, but some people's main interest in ObsPy may be to have a little control over SAC files without the SAC program.  

As far as performance goes, I haven't tested it myself.  For small files, it's probably negligible.  But for longer files, I can see a potential performance and memory penalty for reading in all the data.  I assume there was a reason to program this functionality into the SAC program.

I won't push too hard for this, but I'd like to try to do it.